PsychPython/audio.py

```diff
@@ -64,6 +64,9 @@ def get_devices(device_type=None, device_index=None):
     """Get a list of devices on the current system, optionally of certain types
 
     :param device_type:
+        Can be used to enumerate devices only of a given type: 1=Windows/DirectSound,
+        2=Windows/MME, 11=Windows/WDMKS, 13=Windows/WASAPI, 8=Linux/ALSA,
+        7=Linux/OSS, 12=Linux/JACK, 5=MacOSX/CoreAudio.
```

Comment:
As a bit of guidance wrt. to these selectable audio backends: On macOS, type 5 "CoreAudio" is the only available sound system, ergo the default choice.
On Linux, we default to ALSA for low-level exclusive access to the sound hardware for precise timing, low latency etc.
Linux/OSS is a legacy tech that isn't supported by most Linux distros anymore, so pointless.
Linux/JACK is supported if one wants to share audio devices with other running sound applications, or do some complex post-processing, routing, whatever via JACK. This needs the jack audio connection toolkit installed and enabled, which is easy to do, but not done on default installs of Ubuntu/Debian etc. Few users will need this.
On Windows with current PsychPortAudio, only MME and WASAPI are meaningful choices. MME is the lowest common denominator choice with bad latency and miserable timing, but should work everywhere everytime. It is used if PsychPortAudio('Open',...) explicitely requests latencyclass 0 = no timing precision, no low latency. Good for simple feedback beep tones, sound playback etc. if timing or such does not matter at all. For all higher reqlatencyclass settings, like the default of 1, WASAPI is used for best timing precision and control. It is the best choice on Windows Vista and upwards. WDMKS is nominally supported, but was only a good choice for Windows 2000 / WindowsXP, which we don't support with current PTB anyway - it's there, but its pointless to use. Directsound isn't supported by the portaudio_x64.dll i built, but pointless anyway. Since Vista it is only some compatibility wrapper around WASAPI, and as such worse than WASAPI in any respect and also a source of bugs and crashes, so that's why i omitted it from current driver builds.

Reply:
OK, thanks. That text was lifted verbatim from http://psychtoolbox.org/docs/PsychPortAudio-GetDevices